packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -124,8 +124,10 @@ export class Cluster extends Resource implements ICluster {
       physicalName: props.clusterName,
     });
 
-    const containerInsights = props.containerInsights !== undefined ? props.containerInsights : false;
-    const clusterSettings = containerInsights ? [{ name: 'containerInsights', value: 'enabled' }] : undefined;
+    let clusterSettings = undefined;
+    if (props.containerInsights !== undefined) {
+      clusterSettings = [{ name: 'containerInsights', value: props.containerInsights ? 'enabled' : 'disabled' }];
+    }
```

Comment:
What do you think about constantizing the allowed values for the `containerInsights` setting? i.e.

```
export enum ContainerInsights {
  /**
   * Enable CloudWatch Container Insights will be enabled for the cluster
   */,

  ENABLED = 'enabled',

  /**
   * Disable CloudWatch Container Insights will be enabled for the cluster
   */,
  DISABLED = 'disabled',
}
```

And since the default is to disable container insights, we can not worry about whether `clusterSettings` is undefined so maybe:

```suggestion
    const containerInsights = props.containerInsights !== undefined ? props.containerInsights : false;
    // Explicitly set containerInsights as 'enabled' or 'disabled'
    const containerInsightsSetting = containerInsights ? ContainerInsights.ENABLED : ContainerInsights.DISABLED;
    const clusterSettings = [{ name: 'containerInsights', value: containerInsightsSetting }]


```